common/logging/src/lib.rs

```diff
@@ -256,14 +256,12 @@ pub fn create_tracing_layer(base_tracing_log_path: PathBuf) {
         return;
     };
 
-    let (libp2p_non_blocking_writer, libp2p_guard) = NonBlocking::new(libp2p_writer);
-    let (discv5_non_blocking_writer, discv5_guard) = NonBlocking::new(discv5_writer);
+    let (libp2p_non_blocking_writer, _) = NonBlocking::new(libp2p_writer);
```

Comment:
I think we need the guards for the logging to actually happen. From https://docs.rs/tracing-appender/latest/tracing_appender/non_blocking/index.html

> Note that the WorkerGuard returned by non_blocking must be assigned to a binding that is not _, as _ will result in the WorkerGuard being dropped immediately. Unintentional drops of WorkerGuard remove the guarantee that logs will be flushed during a program’s termination, in a panic or otherwise.

Reply:
💯  nice catch, added back